docs/coding-guidelines/javascript.md

```diff
@@ -367,21 +379,34 @@ if ( object.desired ) { ... }
 
 To test if an object exists and has a property:
 ```js
+// Good:
 if ( object && 'desired' in object ) { ... }
 if ( object && object.desired ) { ... }
+
+// Better, using Lodash's `has` function:
+if ( has( object, 'desired' ) ) { ... }
+
+// Note: 'has' will safely return `false` if a value is missing at any point in the nesting.
+// Even if the chain breaks at 'b' 'has' will return false, rather than throwing an error.
+if ( has( object, 'a.b.c.desired' ) ) { ... }
 ```
 
-Note that the `in` operator checks all inherited properties of an object prototype, which can lead to some unexpected scenarios:
+Note that the `in` operator checks all inherited properties of an object prototype, which can lead to some unexpected scenarios, so should be avoided:
 
 ```js
 'valueOf' in {}; // true
 ```
 
-If testing the presence of a object key using variable input, it's recommended that you use [`Object#hasOwnProperty`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty) instead.
+If testing the presence of a object key using variable input, it's recommended that you use [`Object#hasOwnProperty`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwnProperty) or Lodash's [`has`](https://lodash.com/docs#has) instead.
 
 ```js
-const key = 'valueOf';
-{}.hasOwnProperty( key ); // false
+const key = 'someParam';
+const object = {
+	someParam: 'someValue'
+};
+
+object.hasOwnProperty( key ); // true 
+has( object, key ); // true
```

Comment:
This hunk removes the example of `{}.hasOwnProperty( 'valueOf' )`, though it might be of value to keep it, following the previous note about the `in` operator?